### DISCUSSION TITLE
Bevel Release 1.1.0

### DISCUSSION THREAD

**Discussion by suvajit-sarkar** [hyperledger-labs/blockchain-automation-framework#2597] [Announcements]
## What's new in Bevel
- Platform's with new standalone helm charts, allowing network deployment using just helm commands. 
- Network deployment with proxy or without proxy (single cluster dev/poc setup)
- Network deployment with Hashicorp Vault or without Vault with Kubernetes secrets 

## New Features
- HL Fabric 
  - Network deployment without channel creation
  - Support for custom labels in services and deployments
 
## Deprecation
- No active support for older versions of Bevel (below v1.0.0)
- Support on Kubernetes below v1.23
- No support for Fabric < 2.2.x
  
## Known issues
- Corda Enterprise deployment using firewall support disabled
- 
## Major Bug Fixes
- policy_type variable undefined #2473
- Incorrect storage class name #2513
- RLPx Handshake Error in Quorum Validator Nodes #2563

## Improvements
- All Platform helm charts refactored 
- Hahsicorp Vault paths made flat for easier usage/readability for all platforms

## Documentation
- Update the readme's, setup guide and version information for the DLT platforms
- Update operations guide

## What's Changed
* [fabric] Support add-orderer-organization.yaml playbook by @mgCepeda in https://github.com/hyperledger/bevel/pull/2493
* docs(quorum): operations section under guides tab by @saikumarbommakanti in https://github.com/hyperledger/bevel/pull/2498
* update(docs): fabric configuration and ops sections by @mgCepeda in https://github.com/hyperledger/bevel/pull/2500
* feat(r3corda): enable DLT deployment via GitHub Workflow & Action. by @sailajakommineni in https://github.com/hyperledger/bevel/pull/2499
* update(docs): Substrate configuration file. by @mgCepeda in https://github.com/hyperledger/bevel/pull/2502
* docs(Hyperleder-besu): operations and configurations section under guides tab by @saikumarbommakanti in https://github.com/hyperledger/bevel/pull/2503
* [corda] New charts as per release 1.0.0 by @sownak in https://github.com/hyperledger/bevel/pull/2505
* docs(quorum): configuration section under guides tab by @saikumarbommakanti in https://github.com/hyperledger/bevel/pull/2504
* docs(r3corda): operations section under guide's tab by @sailajakommineni in https://github.com/hyperledger/bevel/pull/2506
* update(docs): Indy configuration and ops sections by @mgCepeda in https://github.com/hyperledger/bevel/pull/2507
* chore: Updated kubeconfig configuration in storage class helm chart by @sownak in https://github.com/hyperledger/bevel/pull/2511
* feat(fabric): Fabric deployment without channel by @mgCepeda in https://github.com/hyperledger/bevel/pull/2515
* feat(besu): implement proposevalidatorvote  method to add or rem… by @saikumarbommakanti in https://github.com/hyperledger/bevel/pull/2517
* [quorum] introduce helm chart deployment capability by @saurabhkumarkardam in https://github.com/hyperledger/bevel/pull/2512
* [quorum] implement istanbul_propose method to add or remove a node by @saurabhkumarkardam in https://github.com/hyperledger/bevel/pull/2516
* Revert "[quorum] introduce helm chart deployment capability" by @suvajit-sarkar in https://github.com/hyperledger/bevel/pull/2519
* [quorum] introduce helm chart deployment capability by @saurabhkumarkardam in https://github.com/hyperledger/bevel/pull/2520
* [corda] Ansible automation with new helm charts by @sownak in https://github.com/hyperledger/bevel/pull/2521
* doc: update roadmap by @suvajit-sarkar in https://github.com/hyperledger/bevel/pull/2522
* feat(besu): updated  image versions for quorum engineering, besu-node and mysql by @saikumarbommakanti in https://github.com/hyperledger/bevel/pull/2524
* [chore] Update flux and redis versions by @sownak in https://github.com/hyperledger/bevel/pull/2526
* (fabric): added support for labels in fabric network by @adityajoshi12 in https://github.com/hyperledger/bevel/pull/2528
* [r3-corda] Cordapps download with secrets by @sownak in https://github.com/hyperledger/bevel/pull/2530
* (fabric): syntax fix for annotations by @adityajoshi12 in https://github.com/hyperledger/bevel/pull/2532
* [fabric] Introduce helm chart deployment capability for version 2.5 by @mgCepeda in https://github.com/hyperledger/bevel/pull/2529
* [indy] Upgrade to 1.12.6 and support for Kubernetes 1.28+ by @sownak in https://github.com/hyperledger/bevel/pull/2531
* [substrate] platform deployment via Helm by @saurabhkumarkardam in https://github.com/hyperledger/bevel/pull/2541
* [fabric] Create configtx and genesis files using just helm by @mgCepeda in https://github.com/hyperledger/bevel/pull/2543
* feat(r3-corda-ent) base network deployment using just helm charts by @suvajit-sarkar in https://github.com/hyperledger/bevel/pull/2542
* [fabric] Catools should be dependency of orderer/peer by @mgCepeda in https://github.com/hyperledger/bevel/pull/2552
* [substrate] update helm charts readmes by @saurabhkumarkardam in https://github.com/hyperledger/bevel/pull/2553
* feat(substrate): enable deployment with proxy and vault via helm by @saikumarbommakanti in https://github.com/hyperledger/bevel/pull/2554
* [indy] Use only helm to deploy Indy network by @sownak in https://github.com/hyperledger/bevel/pull/2555
* feat(r3-corda-ent): helm only deployment using proxy and vault by @suvajit-sarkar in https://github.com/hyperledger/bevel/pull/2559
* [substrate] enable platform deployment via ansible server by @saurabhkumarkardam in https://github.com/hyperledger/bevel/pull/2561
* [fabric] Create channel using just helm by @mgCepeda in https://github.com/hyperledger/bevel/pull/2560
* update(besu,quorum): adapt besu & quorum ansible tasks and template files to charts by @alvaropicazo in https://github.com/hyperledger/bevel/pull/2562
* [fabric] Helm chart updates by @sownak in https://github.com/hyperledger/bevel/pull/2571
* [fabric] Fix Chart versions and add-cli via helm chart by @sownak in https://github.com/hyperledger/bevel/pull/2573
* [fabric] Update add cli and add peer docs by @sownak in https://github.com/hyperledger/bevel/pull/2575
* [fabric] Genesis update and add new channel by @sownak in https://github.com/hyperledger/bevel/pull/2577
* [indy] enable platform deployment via ansible server by @saurabhkumarkardam in https://github.com/hyperledger/bevel/pull/2574
* [shared] support flux latest version 2.3.0 by @saurabhkumarkardam in https://github.com/hyperledger/bevel/pull/2579
* [fabric] Operator readme updates by @sownak in https://github.com/hyperledger/bevel/pull/2580
* [fabric] Update ansible playbooks and roles by @mgCepeda in https://github.com/hyperledger/bevel/pull/2570
* [r3-corda-ent] enable platform deployment via ansible-server by @saikumarbommakanti in https://github.com/hyperledger/bevel/pull/2582
* [fabric] Update ansible playbooks for create a channel by @mgCepeda in https://github.com/hyperledger/bevel/pull/2589
* shared: hot fix by @mgCepeda in https://github.com/hyperledger/bevel/pull/2592
* chore: merged changes from main to develop by @suvajit-sarkar in https://github.com/hyperledger/bevel/pull/2593
* [docs] Updated for latest release by @sownak in https://github.com/hyperledger/bevel/pull/2595
* [chore] release 1.1 merge by @sownak in https://github.com/hyperledger/bevel/pull/2596

**Full Changelog**: https://github.com/hyperledger/bevel/compare/v1.0.1...v1.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/hyperledger/bevel/releases/tag/v1.1.0'>Bevel Release 1.1.0</a>.</em>

**Comment by michaelkhalsa** [comment]
Hi,

Creating a fabric network Single org, 2 peers, 3 orderer, raft using bevel 1.1
Used an external vault (on separate ec2 instance)

Using MacOs (apple silicon) for controller, deployed to an eks cluster.
Ran playbook in sudo, included overrides to set os_aarch and bin directory in cmd.
Added a --kubeconfig location to flux bootstrap which seems to be necessary for flux on apple silicon.
Changed role/check/setup  'Wait for certs to be created in vault'
to use a command instead of shell (so would find vault)
Copied the most recent binaries (was, helm, vault, kubectl, flux) to bevel/bin, so that playbooks would not try to install outdated (and unworkable) versions). Copied over kubectl config, etc., to build.

Ran to completion successfully. Peers up and listening (according to logs), cli installed, orderers running, crypto materials all in vault, genesis block saved to vault and appears correct, everything pushed to the correct GitHub directory for flux.

PLAY RECAP **************************************************************************************************
localhost                  : ok=227  changed=65   unreachable=0    failed=0    skipped=232  rescued=0    ignored=0  

Yet, according to logs, only the system chaincode was installed, i.e., not my chaincode (verified correct GitHub path and permissions). Running peer cli commands, it appears that no 'AllChannel' was created (as specified in network.yaml), nor peers joined to it. I went over network.yaml carefully.

There was also no join-channels job in eks cluster (assume there should be)?

I can run /api/v1/cainfo on the ca server endpoint (when exposed outside cluster) to verify it is live.

Looking through Ansible scripts, I do not see anywhere where the channel is actually created, peers joined, and chaincode installed. These are in the deploy-operator-network script, but not the network-deploy.yaml file.

Is this intentional? Am I missing something? 

Also odd, in the terminal, after completing deploy-network yaml, it then went on to the deploy-operator-network yaml playbook, despite network.env.type not being 'operator' in platform.yaml file. All the tasks were skipped for that anyway. I am looking to see if there is somewhere this variable was changed midstream by one of the scripts.

Thanks for any help, suggestions, or comments.

**Comment by abhayprasadnovartis** [comment]
Hi,

Using the No proxy & No Vault option trying to create a fabric network by following the commands in below link
https://github.com/hyperledger-bevel/bevel/tree/main/platforms/hyperledger-fabric/charts#without-proxy-or-vault

Getting below error at # Join peer to channel and make it an anchorpeer
How do i rectify this error?
useradmin@100:~/hyperLedgerBevel/bevel/platforms/hyperledger-fabric/charts$ helm install peer0-allchannel ./fabric-channel-join --namespace supplychain-net --set global.vault.type=kubernetes
Error: INSTALLATION FAILED: failed post-install: timed out waiting for the condition

Request for the help in this regard
